SwiftWisdom/Core/StandardLibrary/Numbers/Int+Extensions.swift

```diff
@@ -24,3 +24,34 @@ extension IntegerType {
         (0..<self).forEach { _ in closure() }
     }
 }
+
+extension IntegerType {
+    public func ip_toMagnitudeString(decimalPlaces decimalPlaces: Int = 1) -> String {
+        
+        guard self > 999 else { return "\(self)" }
+        
+        let units = ["K", "M", "B", "T", "Q"]
+        
+        let value = Double(self.toIntMax())
+        
+        var magnitude: Int = Int(log10(value) / 3.0) // the order of magnitude of our like count in thousands
+        
+        // divide value by 1000^magnitude to get hundreds value, then round to desired decimal places
+        var roundedHundredsValue = (value / pow(1000.0, Double(magnitude))).ip_roundToDecimalPlaces(decimalPlaces)
```

Comment:
I think breaking out the initial function and then calling round would make this a little easier on the eyes. We have a bit too much logic and parens for a single line